src/Avalonia.Controls/DataValidationErrors.cs

```diff
@@ -29,10 +29,22 @@ public class DataValidationErrors : ContentControl
         /// </summary>
         public static readonly AttachedProperty<bool> HasErrorsProperty =
             AvaloniaProperty.RegisterAttached<DataValidationErrors, Control, bool>("HasErrors");
+        
+        /// <summary>
+        /// Defines the DataValidationErrors.ErrorConverter attached property.
+        /// </summary>
+        public static readonly AttachedProperty<Func<object, object>?> ErrorConverterProperty =
+            AvaloniaProperty.RegisterAttached<DataValidationErrors, Control, Func<object, object>?>("ErrorConverter");
 
+        
         public static readonly StyledProperty<IDataTemplate> ErrorTemplateProperty =
             AvaloniaProperty.Register<DataValidationErrors, IDataTemplate>(nameof(ErrorTemplate));
 
+        /// <summary>
+        /// Defines the DataValidationErrors.DisplayErrors read-only attached property
+        /// </summary>
+        public static readonly AttachedProperty<IEnumerable<object>?> DisplayErrorsProperty =
+            AvaloniaProperty.RegisterAttached<DataValidationErrors, Control, IEnumerable<object>?>("DisplayErrors");
```

Comment:
I wonder if these two ideas make sense:
1. Replace GetExceptionData method with build in converted as ErrorConverterProperty default value (or even when property value is null).
2. Instead of introducing a new property, use existing DataValidationErrors.Errors property to keep already converted values. Having ability to change converter, developers probably won't need both original and mapped values.

Reply:
I tried this in the first step, however I run into 2 issues here: 
1. The converter is set most likely after first evaluation of errors. So any errors in the init state of the VM will not pass the converter. 
2. If you change the converter, one may want to have the errors passing the converter again (consider an App where the user can choose between reduced error messages and full error messages for example). In this case we do not have access to the original errors to pass them again, instead the converted ones will be converted again. 

@maxkatz6 I know you are very experienced and most likely you know way better than me how to achieve this without an additinal read-only property. If you want to, feel free to push changes.

Reply:
In some places we use private/internal avalonia property to keep attached data. Something like that can be done to keep original errors and use it when converted changed.

> If you want to, feel free to push changes.

Sure, I will try to take a look this week.

Reply:
@maxkatz6 ping 
Any thing I can do to keep this PR up?

Reply:
@timunie thank you for reminding me, I completely forgot about this PR :D 

I pushed some changes as long time ago promised:
- Updated samples to include tooltip errors visualization (build in feature, wasn't really documented though)
- Reused DataValidationErrors.SetError in DataGrid
- Changed converter to allow returning null - which supposed to ignore this specific error from being displayed (and counting in HasErrors property)
- Restructured code a bit, used new SetCurrentValue

Reply:
If everything looks good to you after my changed, it can be merged